core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt

```diff
@@ -79,27 +91,148 @@ object GraphQLActionBuilder {
+            else -> {
+                //TODO log warn
+                return
+            }
+        }
+
+        val actionId = "$methodName${idGenerator.incrementAndGet()}"
+
+        //TODO populate
 
+        val params = extractParams(methodName, tableType, kindOfTableType, kindOfTableField, table, tableName)
 
-        val actionId = "$tableName$tableField${idGenerator.incrementAndGet()}"
-        val action = GraphQLAction(actionId, tableName, tableField, tableType)
+        val action = GraphQLAction(actionId, methodName, type, params)
 
         actionCluster[action.getName()] = action
     }
+    private fun extractParams(
+            methodName: String,
+            tableType: String,
+            kindOfTableType: String,
+            kindOfTableField: String?,
+            table: List<Table>,
+            tableName: String
+    ): MutableList<Param> {
+
+        val params = mutableListOf<Param>()
+        val history: Deque<String> = ArrayDeque<String>()
+
+        val gene = getGene(tableType, kindOfTableField, kindOfTableType, table, tableName, history)
+
+        params.add(GQReturnParam(tableType, gene))
+
+
+        return params
+    }
 
+    private fun getGene(
+            tableType: String,
+            kindOfTableField: String?,
+            kindOfTableType: String,
+            table: List<Table>,
+            tableName: String,
+            history: Deque<String> = ArrayDeque<String>()
+    ): Gene {
+
+        when (kindOfTableField) {
+            "LIST" -> {
+                val template = getGene(tableName, kindOfTableType, kindOfTableField, table, tableType, history)
+
+                return ArrayGene(tableName, template)
+            }
+            "OBJECT" -> {
+                return createObjectGene(tableName, tableType, kindOfTableType, table, history)
+            }
+
+            "Int" -> {
+                return IntegerGene(tableName)
+            }
+
+            "String" -> {
+                return StringGene(tableName)
+            }
+            "null"-> {
+                return getGene(tableName, kindOfTableType, kindOfTableField, table, tableType, history)
+            }
+            "Date"-> {
+                return DateGene(tableName)
+            }
+
+            else -> {//see Andrea
+                return StringGene("NotFound")
+            }
+        }
+
+    }
+    private fun createObjectGene(tableName: String,
+                                 tableType: String,
+                                 kindOfTableType: String,
+                                 table: List<Table>,
+                                 history: Deque<String> = ArrayDeque<String>()
+                                ): Gene {
+        val fields :MutableList<Gene> = mutableListOf()
+        for (element in table) {
+            if (element.tableName == tableName) {
+                if(element.kindOfTableType.toString() == "SCALAR") {
+                    val field = element.tableField
```

Comment:
when you do this string comparisons, are we sure that the naming are case sensitive?
if not, should use .equals with ignoreCase

Reply:
Done